<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ScaneliumClass</class>
 <widget class="QMainWindow" name="ScaneliumClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>912</width>
    <height>695</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Scanelium</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <property name="styleSheet">
    <string notr="true"/>
   </property>
   <layout class="QHBoxLayout" name="horizontalLayout">
    <property name="leftMargin">
     <number>0</number>
    </property>
    <property name="topMargin">
     <number>0</number>
    </property>
    <property name="rightMargin">
     <number>0</number>
    </property>
    <property name="bottomMargin">
     <number>0</number>
    </property>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="glWidget" name="bigViewer" native="true">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>600</width>
          <height>300</height>
         </size>
        </property>
        <property name="sizeIncrement">
         <size>
          <width>4</width>
          <height>3</height>
         </size>
        </property>
        <property name="layoutDirection">
         <enum>Qt::LeftToRight</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QToolBox" name="scanTab">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>198</width>
          <height>400</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>230</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="font">
         <font>
          <weight>75</weight>
          <bold>true</bold>
         </font>
        </property>
        <property name="layoutDirection">
         <enum>Qt::LeftToRight</enum>
        </property>
        <property name="styleSheet">
         <string notr="true">QToolBox QScrollArea&gt;QWidget&gt;QWidget {
	background-color: rgb(230, 255, 228);
}
QToolBox::tab {
	background-color: rgb(137, 255, 157);
	border-width: 4;
}
QToolBox {
	background-color: rgb(230, 255, 228);
}
</string>
        </property>
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="initPage">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>213</width>
           <height>565</height>
          </rect>
         </property>
         <attribute name="label">
          <string>Parameters</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout_2">
          <item>
           <widget class="QGroupBox" name="groupBox">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="font">
             <font>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="title">
             <string>Camera</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_4">
             <property name="topMargin">
              <number>9</number>
             </property>
             <item>
              <widget class="QLabel" name="label">
               <property name="text">
                <string>Resolution</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="label_2">
               <property name="text">
                <string>Depth</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="depthResCombo">
               <property name="layoutDirection">
                <enum>Qt::LeftToRight</enum>
               </property>
               <property name="currentIndex">
                <number>1</number>
               </property>
               <item>
                <property name="text">
                 <string>320x240</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>640x480</string>
                </property>
               </item>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="label_3">
               <property name="text">
                <string>Color</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="colorResCombo">
               <property name="layoutDirection">
                <enum>Qt::LeftToRight</enum>
               </property>
               <property name="currentIndex">
                <number>1</number>
               </property>
               <item>
                <property name="text">
                 <string>320x240</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>640x480</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>1280x1024</string>
                </property>
               </item>
              </widget>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_6">
               <item>
                <widget class="QCheckBox" name="recordCheckBox">
                 <property name="text">
                  <string>Record ONI</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QCheckBox" name="recordOnlyBox">
                 <property name="enabled">
                  <bool>false</bool>
                 </property>
                 <property name="text">
                  <string>Only record</string>
                 </property>
                 <property name="checkable">
                  <bool>true</bool>
                 </property>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <widget class="QCheckBox" name="everyframeBox">
               <property name="text">
                <string>Use each depth frame</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="groupBox_2">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="font">
             <font>
              <weight>75</weight>
              <bold>true</bold>
             </font>
            </property>
            <property name="title">
             <string>Scene</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_3">
             <property name="topMargin">
              <number>9</number>
             </property>
             <item>
              <widget class="QLabel" name="sizeLabel">
               <property name="text">
                <string>Size :  cube 1 m</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSlider" name="sizeSlider">
               <property name="minimum">
                <number>5</number>
               </property>
               <property name="maximum">
                <number>50</number>
               </property>
               <property name="singleStep">
                <number>5</number>
               </property>
               <property name="value">
                <number>10</number>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="tickPosition">
                <enum>QSlider::TicksBelow</enum>
               </property>
               <property name="tickInterval">
                <number>5</number>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="gridLabel">
               <property name="text">
                <string>Resolution: 256</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSlider" name="gridSlider">
               <property name="minimum">
                <number>64</number>
               </property>
               <property name="maximum">
                <number>512</number>
               </property>
               <property name="singleStep">
                <number>32</number>
               </property>
               <property name="pageStep">
                <number>64</number>
               </property>
               <property name="value">
                <number>256</number>
               </property>
               <property name="sliderPosition">
                <number>256</number>
               </property>
               <property name="orientation">
                <enum>Qt::Horizontal</enum>
               </property>
               <property name="tickPosition">
                <enum>QSlider::TicksBelow</enum>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QCheckBox" name="doubleYcheckBox">
               <property name="text">
                <string>Double height</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLabel" name="label_6">
               <property name="text">
                <string>Camera location</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QComboBox" name="camposeCombo">
               <property name="layoutDirection">
                <enum>Qt::LeftToRight</enum>
               </property>
               <property name="currentText">
                <string>Face center</string>
               </property>
               <property name="currentIndex">
                <number>1</number>
               </property>
               <item>
                <property name="text">
                 <string>Scene center</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Face center</string>
                </property>
               </item>
               <item>
                <property name="text">
                 <string>Custom</string>
                </property>
               </item>
              </widget>
             </item>
             <item>
              <widget class="QGroupBox" name="customPoseBox">
               <property name="enabled">
                <bool>true</bool>
               </property>
               <property name="title">
                <string>Custom camera pose</string>
               </property>
               <layout class="QVBoxLayout" name="verticalLayout_8">
                <item>
                 <layout class="QHBoxLayout" name="horizontalLayout_3">
                  <item>
                   <widget class="QLabel" name="label_4">
                    <property name="text">
                     <string>Y</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QSlider" name="yAngleSlider">
                    <property name="minimum">
                     <number>-180</number>
                    </property>
                    <property name="maximum">
                     <number>180</number>
                    </property>
                    <property name="orientation">
                     <enum>Qt::Horizontal</enum>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
                <item>
                 <layout class="QHBoxLayout" name="horizontalLayout_4">
                  <item>
                   <widget class="QLabel" name="label_5">
                    <property name="text">
                     <string>X</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QSlider" name="xAngleSlider">
                    <property name="minimum">
                     <number>-180</number>
                    </property>
                    <property name="maximum">
                     <number>180</number>
                    </property>
                    <property name="orientation">
                     <enum>Qt::Horizontal</enum>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
                <item>
                 <layout class="QHBoxLayout" name="horizontalLayout_5">
                  <item>
                   <widget class="QLabel" name="label_7">
                    <property name="text">
                     <string>From center</string>
                    </property>
                   </widget>
                  </item>
                  <item>
                   <widget class="QSlider" name="zDistanceSlider">
                    <property name="maximum">
                     <number>30</number>
                    </property>
                    <property name="value">
                     <number>9</number>
                    </property>
                    <property name="orientation">
                     <enum>Qt::Horizontal</enum>
                    </property>
                   </widget>
                  </item>
                 </layout>
                </item>
               </layout>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
          <item>
           <widget class="QPushButton" name="startButton">
            <property name="text">
             <string>Start reconstruction</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="scanPage">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>108</width>
           <height>152</height>
          </rect>
         </property>
         <attribute name="label">
          <string>Scan</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout_5">
          <item>
           <widget class="QLabel" name="scansizeLabel">
            <property name="text">
             <string>Size: 1m cube</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="scanimagesLabel">
            <property name="text">
             <string>Number of images:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="angleLabel">
            <property name="text">
             <string>Angle diff:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="distLabel">
            <property name="text">
             <string>Distance diff:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="resetButton">
            <property name="text">
             <string>Restart</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="finishButton">
            <property name="text">
             <string>Finish</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer_2">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>40</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="colorTab">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>230</width>
           <height>544</height>
          </rect>
         </property>
         <attribute name="label">
          <string>Color</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout_6">
          <item>
           <widget class="QLabel" name="imagesLabel">
            <property name="text">
             <string>Number of images:</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="iterationslabel">
            <property name="text">
             <string>Iterations: 1</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSlider" name="iterationsSlider">
            <property name="maximum">
             <number>50</number>
            </property>
            <property name="value">
             <number>1</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="tickPosition">
             <enum>QSlider::TicksBelow</enum>
            </property>
            <property name="tickInterval">
             <number>5</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="threadsLabel">
            <property name="text">
             <string>Threads: 4</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QSlider" name="threadsSlider">
            <property name="minimum">
             <number>1</number>
            </property>
            <property name="maximum">
             <number>8</number>
            </property>
            <property name="pageStep">
             <number>1</number>
            </property>
            <property name="value">
             <number>4</number>
            </property>
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="tickPosition">
             <enum>QSlider::TicksBelow</enum>
            </property>
            <property name="tickInterval">
             <number>1</number>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="detailCheckBox">
            <property name="text">
             <string>Increase resolution</string>
            </property>
            <property name="checked">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="imgPyrBox">
            <property name="text">
             <string>Use coarse-to-fine pyramid</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="colorButton">
            <property name="text">
             <string>Start optimization</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="residualLabel">
            <property name="text">
             <string>Current std: 0.00</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="initialResidualLabel">
            <property name="text">
             <string>Initial std: 0.00</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="softStopButton">
            <property name="text">
             <string>Stop</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer_6">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>353</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="SaveTab">
         <property name="geometry">
          <rect>
           <x>0</x>
           <y>0</y>
           <width>98</width>
           <height>47</height>
          </rect>
         </property>
         <attribute name="label">
          <string>Save</string>
         </attribute>
         <layout class="QVBoxLayout" name="verticalLayout_7">
          <item>
           <widget class="QPushButton" name="saveButton">
            <property name="text">
             <string>Save</string>
            </property>
           </widget>
          </item>
          <item>
           <spacer name="verticalSpacer_7">
            <property name="orientation">
             <enum>Qt::Vertical</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>20</width>
              <height>372</height>
             </size>
            </property>
           </spacer>
          </item>
         </layout>
        </widget>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>912</width>
     <height>21</height>
    </rect>
   </property>
   <property name="sizePolicy">
    <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
     <horstretch>0</horstretch>
     <verstretch>0</verstretch>
    </sizepolicy>
   </property>
   <property name="styleSheet">
    <string notr="true">QMenuBar {
	background-color: rgb(230, 255, 228);
}
QMenuBar::item {
	background-color: rgb(230, 255, 228);
}
QMenu {
	background-color: rgb(230, 255, 228);
}
QMenu::item {
	background-color: rgb(230, 255, 228);
}
</string>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuKinfu">
    <property name="title">
     <string>Scanning</string>
    </property>
    <addaction name="actionStart"/>
    <addaction name="actionReset"/>
    <addaction name="actionStop"/>
   </widget>
   <widget class="QMenu" name="menuSettings">
    <property name="title">
     <string>Settings</string>
    </property>
    <addaction name="actionSettings"/>
    <addaction name="actionInfo"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuKinfu"/>
   <addaction name="menuSettings"/>
  </widget>
  <widget class="QStatusBar" name="statusBar">
   <property name="styleSheet">
    <string notr="true">QStatusBar {
	background-color: rgb(230, 255, 228);
}</string>
   </property>
  </widget>
  <action name="actionNew">
   <property name="text">
    <string>New</string>
   </property>
  </action>
  <action name="actionOpen">
   <property name="text">
    <string>Open</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionInit">
   <property name="text">
    <string>Init</string>
   </property>
  </action>
  <action name="actionStart">
   <property name="text">
    <string>Start</string>
   </property>
   <property name="shortcut">
    <string>Space</string>
   </property>
  </action>
  <action name="actionCamera">
   <property name="text">
    <string>Камера</string>
   </property>
  </action>
  <action name="actionKinfu">
   <property name="text">
    <string>Реконструкция</string>
   </property>
  </action>
  <action name="actionColor">
   <property name="text">
    <string>Вычисление цвета</string>
   </property>
  </action>
  <action name="actionMesh">
   <property name="text">
    <string>Mesh</string>
   </property>
  </action>
  <action name="actionClear">
   <property name="text">
    <string>Очистить</string>
   </property>
  </action>
  <action name="actionStop">
   <property name="text">
    <string>Stop</string>
   </property>
   <property name="shortcut">
    <string>Return</string>
   </property>
  </action>
  <action name="actionReset">
   <property name="text">
    <string>Restart</string>
   </property>
   <property name="shortcut">
    <string>R</string>
   </property>
  </action>
  <action name="actionSettings">
   <property name="text">
    <string>Options</string>
   </property>
  </action>
  <action name="actionInfo">
   <property name="text">
    <string>About</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>glWidget</class>
   <extends>QWidget</extends>
   <header>glwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>ScaneliumClass</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>kinfu_begin()</slot>
  <slot>kinfu_stop()</slot>
  <slot>kinfu_run()</slot>
  <slot>kinfu_reset()</slot>
  <slot>save_file()</slot>
  <slot>open_file()</slot>
  <slot>open_settings()</slot>
  <slot>show_info()</slot>
 </slots>
</ui>
